scripts/manage_pes

```diff
@@ -148,456 +89,355 @@ USAGE, OPTIONS, AND EXAMPLES
+
+        sys.stdout.write("\n now writing the new pes list to %s\n" %
+                         newfilename)
+        sys.stdout.write("Please carefully review and/or diff the new file "
+                         "against the\n")
+        sys.stdout.write("original, and if you are satisfied with the "
+                         "changes, move \n")
+        sys.stdout.write("%s to \n" % newfilename)
+        sys.stdout.write("as in with the unix command: \n")
+        sys.stdout.write("mv %s %s\n\n" % (newfilename, pesxml.filename))
+
+        pesxml.write(newfilename)
+
+
+    def queryPes(self):
+        """
+        Read the xml files, query the object, and print the user's choice
+        of output.
+        """
+
+        logger.debug("starting queryPes")
+        filesxml = Files()
+        if self.setby is None:
+            comps = ["allactive", "cam", "cice", "cism", "clm", "drv", "pop"]
+        else:
+            comps = [self.setby]
+        for comp in comps:
+            compsetfile = None
+            compsetfile = filesxml.get_value("COMPSETS_SPEC_FILE",
+                                             attribute={"component":comp})
+            pesxmlfile = filesxml.get_value("PES_SPEC_FILE",
+                                            attribute={"component":comp})
+            pesxml = Pes(infile=pesxmlfile)
+            if not os.path.isfile(pesxmlfile):
+                logger.warning("File %s not found", pesxmlfile)
+                continue
+            if not os.path.isfile(compsetfile):
+                logger.warning("File %s not found", compsetfile)
+                continue
+
+            sys.stdout.write("------------------------------------------- \n")
+            sys.stdout.write(" Pes set by     : %s \n"
+                             % pesxmlfile)
+            sys.stdout.write(" Compsets set by: %s \n" % compsetfile)
+            sys.stdout.write("------------------------------------------- \n\n")
+            matches = self.getMatches(pesxml, grid=self.grid,
+                                      mach=self.machine)
+            self.printMatches(matches)
+
+        logger.debug("exiting queryPes")
+
+    def printMatches(self, gridmatches):
+        for gridnode in gridmatches:
+            comment = None
+            gridname = gridnode.get("name")
+            machnode = gridnode.find("mach")
+            machname = machnode.get("name")
+            pesnode = machnode.find("pes")
+            pesizename = pesnode.get("pesize")
+            compsetname = pesnode.get("compset")
+            commentnode = pesnode.get("comment")
+            sys.stdout.write("grid: %s  machine: %s  compset: %s   "
+                             "pesize: %s \n\n"
+                             % (gridname, machname, compsetname, pesizename))
+            if commentnode:
+                comment = commentnode.text
+            comps = ("atm", "lnd", "ice", "ocn", "rof", "glc", "wav", "cpl")
```

Comment:
General component names (atm, lnd, etc) come from drv_comp.get_valid_model_components()
where drv_comp is :
        drv_config_file = files.get_value("CONFIG_DRV_FILE")
        drv_comp = Component(drv_config_file)